### PR TITLE
added api key authentication

### PR DESCRIPTION
**Pull request by tikisailor**

### CODE REVIEW

**Review by wpmonty — APPROVED** [review]
LGTM

**Review by Joelone — APPROVED** [review]
LGTM

File: src/vote/vote.controller.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import { Body, Controller, Get, Param, Post } from '@nestjs/common';
-import { ApiCreatedResponse, ApiOperation, ApiParam, ApiTags } from '@nestjs/swagger';
+import { Body, Controller, Get, Param, Post, UseGuards } from '@nestjs/common';
+import {ApiCreatedResponse, ApiOperation, ApiParam, ApiSecurity, ApiTags} from '@nestjs/swagger';
 import { VoteMongoService } from './vote.mongo.service';
 import { VoteRequestDto, VoteResponseDto } from './vote.dto';
+import { AuthGuard } from '@nestjs/passport';
 
 @ApiTags('Vote')
+@UseGuards(AuthGuard('api-key'))
```
[the remark this answers is not in the document]
Comment (tikisailor) [reply src/vote/vote.controller.ts:8]: it should be possible to set it global. I'll give it a try. Can also be set on individual endpoints so you can have different strategies on different endpoints

**Comment by tikisailor** [comment]
changed auth to be applied globally and config for heroku deployment

**Pull request merged**